pkg/config/remote/uptane/client.go

```diff
@@ -0,0 +1,245 @@
+		directorLocalStore:  localStoreDirector,
+		directorRemoteStore: newRemoteStoreDirector(targetStore),
+		targetStore:         targetStore,
+	}
+	c.configTUFClient = client.NewClient(c.configLocalStore, c.configRemoteStore)
+	c.directorTUFClient = client.NewClient(c.directorLocalStore, c.directorRemoteStore)
+	return c, nil
+}
+
+// Update updates the uptane client
+func (c *Client) Update(response *pbgo.LatestConfigsResponse) error {
+	c.Lock()
+	defer c.Unlock()
+	err := c.updateRepos(response)
+	if err != nil {
+		return err
+	}
+	err = c.pruneTargetFiles()
+	if err != nil {
+		return err
+	}
+	return c.verify()
+}
+
+// State returns the state of the uptane client
+func (c *Client) State() (State, error) {
+	c.Lock()
+	defer c.Unlock()
+	configRootVersion, err := c.configLocalStore.GetMetaVersion(metaRoot)
+	if err != nil {
+		return State{}, err
+	}
+	directorRootVersion, err := c.directorLocalStore.GetMetaVersion(metaRoot)
+	if err != nil {
+		return State{}, err
+	}
+	configSnapshotVersion, err := c.configLocalStore.GetMetaVersion(metaSnapshot)
+	if err != nil {
+		return State{}, err
+	}
+	return State{
+		ConfigRootVersion:     configRootVersion,
+		ConfigSnapshotVersion: configSnapshotVersion,
+		DirectorRootVersion:   directorRootVersion,
+	}, nil
+}
+
+// Targets returns the current targets of this uptane client
+func (c *Client) Targets() (data.TargetFiles, error) {
+	c.Lock()
+	defer c.Unlock()
+	return c.directorTUFClient.Targets()
+}
+
+// TargetsMeta returns the current raw targets.json meta of this uptane client
+func (c *Client) TargetsMeta() ([]byte, error) {
+	c.Lock()
+	defer c.Unlock()
+	metas, err := c.directorLocalStore.GetMeta()
+	if err != nil {
+		return nil, err
+	}
+	targets, found := metas[metaTargets]
+	if !found {
+		return nil, fmt.Errorf("empty targets meta in director local store")
+	}
+	return targets, nil
+}
+
+type bufferDestination struct {
+	bytes.Buffer
+}
+
+func (b *bufferDestination) Delete() error {
+	return nil
+}
+
+// TargetFile returns the content of a target if the repository is in a verified state
+func (c *Client) TargetFile(path string) ([]byte, error) {
+	c.Lock()
+	defer c.Unlock()
+	err := c.verify()
+	if err != nil {
+		return nil, err
+	}
+	buffer := &bufferDestination{}
+	err = c.configTUFClient.Download(path, buffer)
+	if err != nil {
+		return nil, err
+	}
+	return buffer.Bytes(), nil
+}
+
+func (c *Client) updateRepos(response *pbgo.LatestConfigsResponse) error {
+	err := c.targetStore.storeTargetFiles(response.TargetFiles)
+	if err != nil {
+		return err
+	}
+	c.directorRemoteStore.update(response)
+	c.configRemoteStore.update(response)
+	_, err = c.directorTUFClient.Update()
+	if err != nil {
+		return errors.Wrap(err, "could not update director repository")
+	}
+	_, err = c.configTUFClient.Update()
+	if err != nil {
+		return errors.Wrap(err, "could not update config repository")
+	}
+	return nil
+}
+
+func (c *Client) pruneTargetFiles() error {
```

Comment:
Can you please comment this method explaining what the prune does?
```suggestion
func (c *Client) removeUnusedTargetFiles() error {
```